test/test_torch.py

```diff
@@ -1671,60 +1671,62 @@ def test_remainder_overflow(self):
         self._test_remainder_overflow(self, dtype=torch.int64, device='cpu')
 
     def test_mm(self):
-        # helper function
-        def matrixmultiply(mat1, mat2):
-            n = mat1.size(0)
-            m = mat1.size(1)
-            p = mat2.size(1)
-            res = torch.zeros(n, p)
-            for i, j in iter_indices(res):
-                res[i, j] = sum(mat1[i, k] * mat2[k, j] for k in range(m))
-            return res
-
-        # contiguous case
-        n, m, p = 10, 10, 5
-        mat1 = torch.randn(n, m)
-        mat2 = torch.randn(m, p)
-        res = torch.mm(mat1, mat2)
-
-        res2 = matrixmultiply(mat1, mat2)
-        self.assertEqual(res, res2)
-
-        # non contiguous case 1
-        n, m, p = 10, 10, 5
-        mat1 = torch.randn(n, m)
-        mat2 = torch.randn(p, m).t()
-        res = torch.mm(mat1, mat2)
-
-        res2 = matrixmultiply(mat1, mat2)
-        self.assertEqual(res, res2)
-
-        # non contiguous case 2
-        n, m, p = 10, 10, 5
-        mat1 = torch.randn(m, n).t()
-        mat2 = torch.randn(m, p)
-        res = torch.mm(mat1, mat2)
-
-        res2 = matrixmultiply(mat1, mat2)
-        self.assertEqual(res, res2)
-
-        # non contiguous case 3
-        n, m, p = 10, 10, 5
-        mat1 = torch.randn(m, n).t()
-        mat2 = torch.randn(p, m).t()
-        res = torch.mm(mat1, mat2)
-
-        res2 = matrixmultiply(mat1, mat2)
-        self.assertEqual(res, res2)
-
-        # test with zero stride
-        n, m, p = 10, 10, 5
-        mat1 = torch.randn(n, m)
-        mat2 = torch.randn(m, 1).expand(m, p)
-        res = torch.mm(mat1, mat2)
-
-        res2 = matrixmultiply(mat1, mat2)
-        self.assertEqual(res, res2)
+        def _test_mm(n, m, p, dtype, genf):
```

Comment:
It could be worth adding test coverage for torch.mm's out flag to explicitly exercise the _out variant of this function.

Reply:
Could you show me more details on how to exercise the _out variant in the torch.mm. I will try to figure it out tomorrow as well.

Reply:
By calling e.g. torch.mm(a, b, out=c), where c has been preallocated.

Reply:
Add two additional cases in test_mm() to explicitly exercise the _out variant in torch.mm()